components/validator/guardian.go

```diff
@@ -164,59 +162,68 @@ func (g *Guardian) handleValidationRequested(ctx context.Context) {
 	}
 }
 
-// TODO(pangssu): Retry a failed or missed transaction.
+func (g *Guardian) tryConfirmTransaction(ctx context.Context, event *bindings.SecurityCouncilValidationRequested) error {
+	cCtx, cCancel := context.WithTimeout(ctx, g.cfg.NetworkTimeout)
+	callOpts := utils.NewCallOptsWithSender(cCtx, g.cfg.TxManager.From())
+	isConfirmed, err := g.securityCouncilContract.IsConfirmed(callOpts, event.TransactionId)
+	cCancel()
+	if err != nil {
+		g.log.Error("failed to get confirmation", "transactionId", event.TransactionId, "err", err)
+		return err
+	}
+
+	if isConfirmed {
+		g.log.Info("this transaction has already been confirmed", "transactionId", event.TransactionId)
+		return nil
+	}
+
+	isValid, err := g.ValidateL2Output(ctx, event.OutputRoot, event.L2BlockNumber.Uint64())
+	if err != nil {
+		g.log.Error("failed to verify the output to be replaced", "transactionId", event.TransactionId, "err", err)
+		return err
+	}
+
+	if isValid {
+		g.log.Info("the output to be replaced is valid")
+
+		cCtx, cCancel := context.WithTimeout(ctx, g.cfg.NetworkTimeout)
+		tx, err := g.ConfirmTransaction(cCtx, event.TransactionId)
+		cCancel()
+		if err != nil {
+			g.log.Error("failed to create confirm tx", "transactionId", event.TransactionId, "err", err)
+			return err
+		}
+		if tx == nil {
+			return nil
+		}
+
+		if err := g.sendTransaction(ctx, tx); err != nil {
+			return err
+		}
+	} else {
+		g.log.Info("ignore this challenge because the output to be replaced is invalid")
+		return nil
+	}
+
+	return nil
+}
+
 func (g *Guardian) processOutputValidation(ctx context.Context, event *bindings.SecurityCouncilValidationRequested) {
 	ticker := time.NewTicker(10 * time.Second)
 	defer func() {
 		ticker.Stop()
 		g.wg.Done()
 	}()
 
-	for {
-	Loop:
+	for ; ; <-ticker.C {
 		select {
-		case <-ticker.C:
-			cCtx, cCancel := context.WithTimeout(ctx, g.cfg.NetworkTimeout)
-			callOpts := utils.NewCallOptsWithSender(cCtx, g.cfg.TxManager.From())
-			isConfirmed, err := g.securityCouncilContract.IsConfirmed(callOpts, event.TransactionId)
-			cCancel()
-			if err != nil {
-				g.log.Error("failed to get confirmation", "transactionId", event.TransactionId, "err", err)
-				break Loop
-			}
-
-			if isConfirmed {
-				g.log.Info("this transaction has already been confirmed", "transactionId", event.TransactionId)
-				return
-			}
-
-			isValid, err := g.ValidateL2Output(ctx, event.OutputRoot, event.L2BlockNumber.Uint64())
+		case <-ctx.Done():
+			return
+		default:
+			err := g.tryConfirmTransaction(ctx, event)
```

Comment:
With adding retries when a transaction fails to send, we can revert the changes in this [PR](https://github.com/kroma-network/kroma/pull/95).